test/cpp/end2end/grpclb_end2end_test.cc

```diff
@@ -237,7 +237,7 @@ class BalancerServiceImpl : public BalancerService {
               .num_calls_finished_with_client_failed_to_send();
       client_stats_.num_calls_finished_known_received +=
           request.client_stats().num_calls_finished_known_received();
-      for (const auto& drop_token_count :
+      for (const auto drop_token_count :
```

Comment:
I am not sure whether I should delete the `&` here, but I guess yes.

Reply:
I think the `&` is fine here, because the protobuf `calls_finish_with_drop()` method returns a reference to a `RepeatedPtrField<>`, so its elements will remain existing as long as the protobuf object does.

Reply:
I see. I should review that totw.

Reply:
In fact it's always valid: binding a temporary object to a reference to const on the stack lengthens the lifetime of the temporary to the lifetime of the reference itself: see https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/

Reply:
Yes. After reading both the internal totw and the external gotw, I think maybe it's generally better to avoid the `&` in such cases. The reason is detailed in the totw. In short, using a `&` here doesn't improve efficiency (because of return-value-optimization, which moves (instead of copies) the returned object to initialize the LHS object), but makes code reviewers consider more conditions to determine whether it's legit or not.

Reply:
I don't think it's a problem in this case.  It's only in assignments where the scope of both sides may be different where it matters.